cpp/velox/memory/VeloxColumnarBatch.cc

```diff
@@ -147,7 +168,7 @@ std::shared_ptr<VeloxColumnarBatch> VeloxColumnarBatch::select(
   std::vector<VectorPtr> childVectors;
   childNames.reserve(columnIndices.size());
   childVectors.reserve(columnIndices.size());
-  auto vector = getFlattenedRowVector();
+  auto vector = getPartialFlattenedRowVector(columnIndices);
```

Comment:
I don't know why we need to flatten it, can we remove it?

Reply:
I'm afraid not. `ColumnarBatch.select` are used in several scenario, including `ColumnarPartialProject` and `ColumnarCachedBatchSerializer`. 
The former will do `ensureFlatten` again before converting to arrow by `VeloxColumnarBatch::exportArrowArray`. But since at that time, the input batch is already flattend, so there is no cost.
The latter will access the bytes of input vector directly, so we cannot remove this.

Reply:
For the former, we can do the flatten for essential columns when converting to arrow.
For the latter, it serializes the batch to bytes, does the serializer only supports flatten batch? I remember PrestoVectorSerde supports all the RowVector.

Reply:
For the latter, it use `UnsafeRowSerializer`, and internally use `DecodedVector` to decode input row vector, while the `DecodedVector` requires that the input should be flat, constant or dictionary, but don't support lazy vector.

Reply:
So we only need to load lazy vector here.

Reply:
udpate. remove flatten in `ColumnarBatch.select`
1. for the former case, `exportArrowArray` will make sure it's flattend.
2. for the latter case, flatten it before `UnsafeRowSerializer`.

Reply:
You notice me something, maybe we can support the dictionary vector in columnar to row in the future